src/main/scala/io/iohk/ethereum/blockchain/sync/FastSync.scala

```diff
@@ -539,6 +553,11 @@ class FastSync(
     private def insertBlocks(requestedHashes: Seq[ByteString], blockBodies: Seq[BlockBody]): Unit = {
       (requestedHashes zip blockBodies).foreach { case (hash, body) =>
         blockchain.save(hash, body)
+        blockSyncCache.get(hash, () => (
+          Try(blockchain.getBlockHeaderByHash(hash)).toOption.flatten,
```

Comment:
Why this `Try` , are option is not enough?  (also there is possibility that header is in cache so maybe you could get header from cache or db like in header validation)

Reply:
I provided Try because getBlockHeaderByHash throws an exception

Reply:
i think we never catch exceptions from `blockchain` methods,  in order to quickly know if there is somethig wrong with db.

Reply:
I think the problem was when I was getting some serialization/deserialization errors. It crushed the whole app